pkg/cmd/infra/deployer/deployer.go

```diff
@@ -240,7 +238,7 @@ func (d *Deployer) Deploy(namespace, rcName string) error {
 			continue
 		}
 		// Skip the deployment if it's already scaled down.
-		if candidate.Spec.Replicas == 0 {
+		if *candidate.Spec.Replicas == 0 {
```

Comment:
`candidate.Spec.Replicas == nil || candidate.Spec.Replicas == 0` or we just assume it is set? In a sane world this would be probably always set (by admission) so when you get existing object this is present.

Reply:
i will add a check to be sure